### ISSUE TITLE
[Bug Report] Reduce operations for ndim > 2 do not work for certain shapes

### ISSUE THREAD

**Issue by pglusacTT** [tenstorrent/tt-metal#15996]
**Describe the bug**
Doing reduce operations with certain shapes produce incorrect results.
This behavior was observed only for ndim > 2 and for certain shapes, not for all of them.

I did a search of different shapes for ndim=3 and ndim=4 and for a single dim < 10 and tried reducing them in the similar fashion as the code below. Here are the results:

For ndim=4: Failed for 14.71% of the tests
For ndim=3: Failed for 2.6% of the tests

**To Reproduce**
```python
import torch
import ttnn

def test(shape, device):
    tensor = torch.ones(shape)
    tt_tensor = ttnn.from_torch(tensor, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
    tt_out = ttnn.sum(tt_tensor)
    tt_out = ttnn.to_torch(tt_out)
    torch_out = torch.sum(tensor)

    if not torch.allclose(torch_out, tt_out, rtol=1e-3):
        print(f"{tt_out=}")
        print(f"{torch_out=}")
        print("-" * 10)

if __name__ == "__main__":
    device_id = 0
    device = ttnn.open_device(device_id=device_id)
    shapes = [(9, 5, 7), (7, 7, 7), (10, 10, 10, 10)]
    for shape in shapes:
        test(shape, device)

    ttnn.close_device(device)
```

The code outputs:
```
tt_out=TorchTensor([[[316.]]])
torch_out=tensor(315.)
----------
tt_out=TorchTensor([[[344.]]])
torch_out=tensor(343.)
----------
tt_out=TorchTensor([[[[9984.]]]])
torch_out=tensor(10000.)
----------
```

**Expected behavior**
Sum in this case should equal to the product of dim sizes and it isn't.

**Environment information:**
- OS: Ubuntu 22.04
- Version of software: dde5614e816b20110b0bb8c96aeefe5a0a7bada1

**Comment by bbradelTT** [comment]
@pglusacTT @vladimirjovanovicTT can one of you please test on latest main and report if that works for you or not.

This test passes for me after I fixed https://github.com/tenstorrent/tt-metal/issues/12662 via https://github.com/tenstorrent/tt-metal/pull/16925

**Comment by bbradelTT** [comment]
Since I haven't heard anything and the reproduction code does the right thing for me now, closing.
Please re-open and provide more information if you still see the issue.
